### DISCUSSION TITLE
facing one issue

### DISCUSSION THREAD

**Discussion by pratik-2707** [community/community#60826] [New to GitHub]
what can i do if i already made a pr
but now i have to delete the pr  and make new pr and include some changes only from all those changes

**Comment by the-officialjosh** [comment]
If you have already made a pull request (PR) but need to delete it and create a new PR with only specific changes, you can follow these steps:

1. Close the existing PR: Go to the repository where the PR was created and find the existing PR. Typically, there will be an option to close the PR. Closing the PR does not delete it, but it marks it as closed and prevents further changes from being made to it.

2. Create a new branch: Checkout to a new branch in your local repository where you can make the desired changes without affecting the existing branch associated with the closed PR.

3. Cherry-pick the specific changes: In the new branch, selectively apply the changes you want to include in the new PR. You can use the `git cherry-pick` command to choose specific commits or changesets and apply them to your current branch.

4. Push the new branch: Once you have made the necessary changes, push the new branch to your remote repository.

5. Create a new PR: On the remote repository, create a new PR using the newly pushed branch. Include a description explaining the changes you made and any context or relevant information.

By following these steps, you can effectively delete the existing PR and create a new one with only the specific changes you want to include.

**Comment by pratik-2707** [comment]
okay and what if i want to choose specific changes from my stashed changes

**Reply by the-officialjosh** [reply]
If you have stashed changes and you want to selectively apply only certain changes from the stash, you can use the `git stash apply` or `git stash branch` commands along with the appropriate options. Here's what you can do:

1. List your stashed changes: Use the command `git stash list` to see a list of your stashed changes. It will display a list of stash entries along with their corresponding index or name.

2. Apply the stash with partial changes: You can use `git stash apply --patch <stash>` to selectively apply changes from a specific stash entry. Replace `<stash>` with the index or name of the stash entry you want to apply.

   This command will start an interactive session where you can choose which changes you want to apply from the stash. You'll be presented with a diff hunk and prompted with options like `y` (apply this hunk), `n` (do not apply this hunk), `q` (quit), and more. Select the desired changes you want to include and proceed accordingly.

3. Create a new branch or commit: Once you have applied the desired changes, you can create a new branch or commit with the modified code.

   - To create a new branch with the changes, use `git checkout -b new-branch-name`.
   - To commit the changes directly, use `git commit -m "Commit message"`.

Remember to choose a meaningful branch or commit name to reflect the changes you have applied.

By selectively applying changes from your stash, you can include specific modifications in a new branch or commit without bringing in all the changes stored in the stash.

**Comment by pratik-2707** [comment]
if possible i would like to discuss this in google meet .
